WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderDetailFeaturedImageView.swift

```diff
@@ -72,13 +73,29 @@ class ReaderDetailFeaturedImageView: UIView, NibLoadable {
     /// Listens for contentOffset changes to track when the user scrolls
     private var scrollViewObserver: NSKeyValueObservation?
 
-    /// Stores the nav bar appearance before we change it to transparent
-    /// this allows us to reset it when the view disappears
-    private var originalNavBarAppearance: NavBarAppearance?
+    /// Flag indicating whether the navigation bar appearance needs restoration when the view disappears. Default is `false`.
+    ///
+    /// Flag changes to `true` when this view changes the navigation bar tint color. In other words, when the `navBarTintColor` setter is called.
+    private var navigationBarAppearanceNeedsRestoration = false
 
-    private var navBarTintColor: UIColor = Styles.endTintColor {
-        didSet {
-            navigationBar?.setItemTintColor(useCompatibilityMode ? .textInverted : navBarTintColor)
+    /// The original navigation bar tint color.
+    ///
+    /// This property is used to restore the navigation bar tint color to its original value.
+    private var previousNavBarTintColor: UIColor?
```

Comment:
This is removed in https://github.com/wordpress-mobile/WordPress-iOS/pull/20363/commits/4543fedb1a85f8e204a264e7e39fd266efe39c70